### ISSUE TITLE
No support for horizontal scroll wheel

### ISSUE THREAD

**Issue by lkd70** [atom/atom#12227]
### Prerequisites
- [x] Can you reproduce the problem in [safe mode](http://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode)?
- [x] Are you running the [latest version of Atom](http://flight-manual.atom.io/hacking-atom/sections/debugging/#update-to-the-latest-version)?
- [x] Did you check the [debugging guide](http://flight-manual.atom.io/hacking-atom/sections/debugging/)?
- [x] Did you check the [FAQs on Discuss](https://discuss.atom.io/c/faq)?
- [x] Are you reporting to the [correct repository](https://github.com/atom/atom/blob/master/CONTRIBUTING.md#atom-and-packages)?
- [x] Did you [perform a cursory search](https://github.com/issues?q=is%3Aissue+user%3Aatom+-repo%3Aatom%2Felectron) to see if your bug or enhancement is already reported?

For more information on how to write a good [bug report](https://github.com/atom/atom/blob/master/CONTRIBUTING.md#how-do-i-submit-a-good-bug-report) or [enhancement request](https://github.com/atom/atom/blob/master/CONTRIBUTING.md#how-do-i-submit-a-good-enhancement-suggestion), see the `CONTRIBUTING` guide.
### Description

At the time being there's no support for a horizontal scroll wheel, something many programmers look for in a mouse these days. 
### Steps to Reproduce
1. Open Atom
2. Attempt to use horizontal scroll wheel on side of mouse
3. Sit and wonder if the mouse is broken for a good 2 minutes
4. Blame the tools for it not working

**Expected behavior:** page will horizontally scroll.

**Actual behavior:** Nothing
### Versions

Atom     : 1.8.0
Electron : 0.36.8
Chrome : 47.0.2526.110
Node     : 5.1.1
OS         : Windows 10 Pro

**Comment by DallasO** [comment]
Do you mean a dedicated horizontal scroll? Has worked fine for me pre-1.8.0 by pushing my vertical scroll to the sides.

**Comment by lkd70** [comment]
I do indeed, sorry for not being too clear.

I mean an additional scroll wheel dedicated to horizontal scrolling. Like the one featured on the Logitech MX Master (the mouse I use).

**Comment by 50Wliu** [comment]
Thanks for the clarification @lkd70.  Does the horizontal scroll wheel work in Chrome?

**Comment by lkd70** [comment]
It works perfectly fine in chrome browser.

**Comment by alexcode9** [comment]
same for me, Logitech MX master horizontal scroll wheel working almost everywhere but not in Atom

**Comment by quicklikerabbit** [comment]
I'm having a similar issue; horizontal scroll doesn't work when you first click to the left or right, but if you hold down scrolling starts to occur. Scrolling speed is controlled by the 'Acceleration' attribute in the logitech control center. Still not ideal, but at least the scrolling works to some degree. I'm using a Logitech M705 on MacOS Sierra.

**Comment by senyaak** [comment]
Get same problem after update. It has worked in older version(

**Comment by Kelendria** [comment]
Like alexcode9, it works intermittently after using the thumb wheel in Chrome. I'm using the Logitech MX Master as well. Unlike alexcode9, the horizontal scroll only works for one tick after using the thumb wheel in Chrome.

**Comment by ndrs92** [comment]
Can reproduce this issue. Also works for me if I use the Chrome trick.

I also have a Logitech MX Master.

**Comment by tom-sherman** [comment]
Also reproduced the issue on 1.16.0 with my MX Master. Chrome workaround works for me too

**Comment by Ixous** [comment]
Same issue. Atom is the only program where the side scroll doesn't work. Also with an MX Master.
This is a huge problem..
The chrome fix works but only for a while, less long if atom is in fullscreen... And there is no way to make it work when using two full screen atom window.

**Comment by KvalheimRacing** [comment]
Any news here? horizontal scrolling just stopped working. was working before, got the newest version of atom. Same mouse

**Comment by senyaak** [comment]
For me the problem gone away for a long time) some update done this)

**Comment by lkd70** [comment]
Issue still exists.

**Comment by KvalheimRacing** [comment]
@senyaak I think this issue is related to #5716 Can you go from another application than chrome (try word or notepad) and then directly to atom, and horizontal scrolling works? If so, we need to find out what our differences are. My atom version is 1.18.0. Remember having a older atom at one point, and never having an issue with this horizontal scrolling. Don't know what that version was.

**Comment by KvalheimRacing** [comment]
I just bought a new logitech t630, it also has horizontal scrolling (in touch, not as a wheel). Tested everything now, simultanious as the MX Master, having the same problems with the MX Master, but everything works with the t630. What do you guys think, is it a software problem with the logitech MX Master or is it atom because of different scrolling method?

**Comment by KvalheimRacing** [comment]
This is the same issue as https://github.com/atom/atom/issues/5716#issuecomment-314487771. people having trouble with this should check it out. We found out it is most likely a Chrome bug. I filed a bug [here](https://bugs.chromium.org/p/chromium/issues/detail?id=746954), please star if you would like to get this bug fixed faster.

**Comment by boaz-amit** [comment]
@KvalheimRacing I never had any problems with horizontal scrolling in Chrome using the MX Master.

**Comment by KvalheimRacing** [comment]
hmh, wierd... Any thoughts @50Wliu or @nathansobo?

**Comment by KvalheimRacing** [comment]
@lkd70 @Ixous @tom-sherman @ndrs92 @Kelendria @alexcode9, do you still have this issue (if not using the chrome trick)?

**Comment by Ixous** [comment]
Yes I do still have this issue (still an MX Master, maybe this is a Logitch issue)

**Comment by Ixous** [comment]
And I have never had issues with the side scroll on Chrome thus I doubt this is the root of the problem. Although Atom uses an older version of Chrome I think

**Comment by boaz-amit** [comment]
@KvalheimRacing  I did not use Logitech Setpoint with my previous mice. I do use Logitech Options with my current MX Master. Horizontal scrolling does not work in Atom with this mouse since I got it. Thanks.

**Comment by KvalheimRacing** [comment]
@boaz-amit ok then we have the same software. I assume you are using *Application specific settings* in logitech options, and then have choosen Atom, and then horizontal scrolling on the thumb wheel? Also, have you tried the "Chrome trick" explained in the comments over? 

> EDIT: I don't know how and why but I found for me a solution... open atom, click on a chrome page and scroll the horizontal wheel, return to atom, now it's working

**Comment by boaz-amit** [comment]
@KvalheimRacing I am not using any application-specific settings in Logitech Options. As for the Chrome trick, for me horizontal scrolling in Chrome and switching to Atom causes the horizontal scroll to work for one scroll movement Atom and then it does not respond anymore. Switch back to Chrome, the scroll continues to work as expected.

**Comment by KvalheimRacing** [comment]
@boaz-amit hmm, ok. What version of atom do you have?

**Comment by boaz-amit** [comment]
@KvalheimRacing Currently, 1.18.0

**Comment by KvalheimRacing** [comment]
Ok (I have the same) then I can't see any other clear patterns than that we are using atom, and the mx master mouse. Starting to belive this may be a logitech issue in some way.. @50Wliu, @nathansobo, any ideas or suggestions to move forward here?

**Comment by nathansobo** [comment]
I'm sort of at a loss here if this is a bug in Chrome. We base our horizontal scrolling on the `mousewheel` event provided by Electron. If we get an event with a `deltaX`, we'll honor it, but it sounds like we're not getting that event for some reason. @ungb do you want to purchase and expense one of these Logitech MX Master mice and see if we can learn more with the actual hardware?

**Comment by lkd70** [comment]
Just as a side-note, it works fine in VisualStudio Code, which too is built on electron iirc.

**Comment by nathansobo** [comment]
I wonder if they are using `mousewheel` events or some other mechanism for capturing scroll.

**Comment by boaz-amit** [comment]
@nathansobo Could this be related to the changes introduced in #2567 a few years ago?

**Comment by nathansobo** [comment]
I don't think so. It's something deeper happening at the event level. Until we get hold of one of these mice or someone with one of these mice is interested in digging deeper, we're stuck. I'll talk internally about acquiring one.

**Comment by lkd70** [comment]
I'm more than willing g to provide the testing needed, though I dont use atom anymore I am happy to help out. Though I do recommend getting one of the mice, they're fantastic.

**Comment by nathansobo** [comment]
@ungb Did you say you were unable to reproduce this with one of these mice?

**Comment by ungb** [comment]
@nathansobo I got the mouse and tried it on my windows machine and mac and saw that horizontal scrolling works.  


The mouse has different ways of connecting.  Bluetooth or dongle.  I've tried both and didn't see any issues.

@lkd70 is there any other info you can provide regarding this issue?  How is the mouse connected? 
Also, did you install any additional softwares or drivers besides the one that windows uses when you connect?  


OS I tested on:
Windows 10 Pro Creators Update x64
Mac 10.12.6


I was using this mouse to test: https://www.amazon.com/Logitech-Master-Wireless-Mouse-High-precision/dp/B00TZR3WRM

**Comment by lkd70** [comment]
@ungb Thanks for testing it out. 

Here's some additional information as requested:

Connection method: USB dongle (as provided by Logitech)
Additional software/drivers: Logitech Options (should come with the mouse)

Logitech Options -> Thumb Wheel direction: Inverted (though I've tried both)
Logitech Options -> Thumb Wheel Sensitivity: 100% (default also fails)

Thanks again for reviewing this.

**Comment by KvalheimRacing** [comment]
@ungb this is wierd, I am starting to think it could be a internal software fault from logitech. Wanna switch mouses? x)

**Comment by boaz-amit** [comment]
@ungb Thanks for testing. This indeed appears to suggest that the fault
lies with Logitech Options. Since this software is very much needed to get
the most out of the mouse, can you please try installing it?

**Comment by ungb** [comment]
Sorry to respond to this so late.  I just tried it with the software and uninstalled it.  Atom stops working with the software Logitech Options installed and starts working again once you uninstall it.  

I didn't try it with any of the settings.  Maybe there's something in there disabling it?  

I'm going to close this issues since the fix is to uninstall Logitech Options :/.  I contacted Logitech support to make them aware of this issue.  If I find something else I'll update here.
